ipn/local.go

```diff
@@ -427,6 +429,27 @@ func (b *LocalBackend) updateFilter(netMap *controlclient.NetworkMap) {
 	b.e.SetFilter(filter.New(netMap.PacketFilter, localNets, b.e.GetFilter(), b.logf))
 }
 
+// updateDNSMap updates the domain map in the DNS resolver in wgengine
+// based on the given netMap and user preferences.
+func (b *LocalBackend) updateDNSMap(netMap *controlclient.NetworkMap) {
+	if netMap == nil {
+		return
+	}
+	dnsMap := &tsdns.DNSMap{DomainToIP: make(map[string]netaddr.IP)}
+	for _, peer := range netMap.Peers {
+		if len(peer.Addresses) == 0 {
+			continue
+		}
+		domain := peer.Hostinfo.Hostname
+		// Like PeerStatus.SimpleHostName()
+		domain = strings.TrimSuffix(domain, ".local")
+		domain = strings.TrimSuffix(domain, ".localdomain")
+		domain = domain + ".ipn.dev"
```

Comment:
This is fine for this PR, but we will have to rethink this soon for workgroups. I believe we intend to change domains for that though, so this is fine.